doc/source/glossary.rst

```diff
@@ -0,0 +1,179 @@
+
+.. _Offset:
+
+Offset
+    A numeric value that identifies a distance within a group of bytes, to uniquely identify a single byte, or the
+    start of a run of bytes.  This is often relative (offset from another object/item) but can be absolute (offset from
+    the start of a region of data).
+
+P
+-
+.. _Packed:
+
+Packed
+    Structures are often :ref:`aligned<alignment>` meaning that the various members (subtypes) are always aligned at
+    particular values (usually multiples of 2, 4 or 8).  Thus if a particular value is an odd number of bytes, the
+    next chunk of data containing useful information would start at an even offset, and a single byte of
+    :ref:`padding<padding>` would be used to ensure appropriate :ref:`alignment<alignment>`.  In packed structures, no
+    padding is used, and offsets may be at odd offsets.
+
+.. _Padding:
+
+Padding
+    Data that (usually) contains no useful information.  The typical value used for padding is 0, so should a string
+    :ref:`object<object>` that has been allocated a particular number of bytes, contain a string of fewer bytes, the remaing bytes
+    will be padded with null (0) bytes.
+
+.. _Page:
+
+Page
+    A specific chunk of contiguous data.  It is an organizational quantity of memory (usually 0x1000, or 4096 bytes).
+    Pages, like pages in a book, make up the whole, but allow for specific chunks to be allocated and used as necessary.
+    Operating systems uses pages as a means to have granular control over chunks of memory.  This allows them to be
+    reordered and reused as necessary (without having to move large chunks of data around), and allows them to have
+    access controls placed upon them, limiting actions such as reading and writing.
+
+.. _Page Table:
+
+Page Table
+    A table that points to a series of :ref:`pages<page>`.  Each page table is typically the size of a single page,
+    and page tables can point to pages that are in fact other page tables.  Using tables that point to tables, it's
+    possible to use them as a way to map a particular address within a (potentially larger, but sparsely populated)
+    virtual space to a concrete (and usually contiguous) physical space, through the process of :ref:`mapping<map>`.
+
+.. _Pointer:
+
+Pointer
+    A value within memory that points to a different area of memory.  This allows objects to contain references to
+    other objects without containing all the data of the other object.  Following a pointer is known as :ref:`dereferencing<dereference>`
+    a pointer.  Pointers are usually as large as the size of the
+
+R
+-
+.. _Range:
+
+Range
+    This is the grouping the output values for a mapping or mathematical function.
+
+S
+-
+.. _Struct:
+
+Struct, Structure
+    A means of containing multiple different :ref:`type<types>` associated together.  A struct typically contains
+    other :ref:`type<types>`, one directly after another (unless :ref:`packing<packed>` is involved).  In this way
```

Comment:
Isn't `directly after another` exactly what packing does?

Reply:
It is, I've reworded it a little.  I can't tell whether structs are typically packed or not, but hopefully my correction clarifies that alignment and packing are somewhat opposite ends of the spectrum.